modules/evm/src/lib.rs

```diff
@@ -861,75 +909,41 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
-	/// Get the account basic in EVM format.
-	pub fn account_basic(address: &EvmAddress) -> Account {
-		let account_id = T::AddressMapping::get_account_id(address);
-
-		let nonce = Self::accounts(address).map_or(Default::default(), |account_info| account_info.nonce);
-		let balance = T::Currency::free_balance(&account_id);
-
-		Account {
-			nonce: U256::from(UniqueSaturatedInto::<u128>::unique_saturated_into(nonce)),
-			balance: U256::from(UniqueSaturatedInto::<u128>::unique_saturated_into(balance)),
-		}
-	}
-
-	/// Get code hash at given address.
-	pub fn code_hash_at_address(address: &EvmAddress) -> H256 {
-		if let Some(AccountInfo {
-			contract_info: Some(contract_info),
-			..
-		}) = Self::accounts(address)
-		{
-			contract_info.code_hash
-		} else {
-			code_hash(&[])
-		}
-	}
-
-	/// Get code at given address.
-	pub fn code_at_address(address: &EvmAddress) -> BoundedVec<u8, T::MaxCodeSize> {
-		Self::codes(&Self::code_hash_at_address(address))
-	}
-
-	pub fn update_contract_storage_size(address: &EvmAddress, change: i32) {
-		if change == 0 {
+	/// Create an account.
+	/// - Create new account for the contract.
+	/// - Update codes info.
+	/// - Update maintainer of the contract.
+	/// - Save `code` if not saved yet.
+	pub fn create_account(source: H160, address: H160, code: Vec<u8>) {
+		let bounded_code: BoundedVec<u8, T::MaxCodeSize> = code
+			.try_into()
+			.expect("checked by create_contract_limit in ACALA_CONFIG; qed");
+		if bounded_code.is_empty() {
 			return;
 		}
-		ContractStorageSizes::<T>::mutate(address, |val| {
-			if change > 0 {
-				*val = val.saturating_add(change as u32);
-			} else {
-				*val = val.saturating_sub((-change) as u32);
-			}
+
+		// if source is account, the maintainer of the new contract is source.
+		// if source is contract, the maintainer of the new contract is the maintainer of the contract.
+		let maintainer = Self::accounts(source).map_or(source, |account_info| {
```

Comment:
I'm not sure about this behaviour.  If someone uses a contract to deploy a contract then he loses the ownership.

Reply:
Refer to this document to achieve
```
Contract Maintainer
The deployer of a contract will become the maintainer of the contract
If a contract deploy another contract, the maintainer of the newly deployed contract is the maintainer of the deployer contract
Maintainer can transfer the role to other accounts, which can be a smart contract
```

Reply:
@zjb0807 I understand that, I am questioning if that's the behaviour we want

Reply:
I don't see issue here. When a parent contract deploy a child contract, the ownership of child contract should by default belong to parent contract. It is up to parent contract to decide if it want to transfer the ownership of child contract to others.